### PR TITLE
dev-libs/libsigc++: properly disable benchmarking (bug 599764) and ma…

### PR DESCRIPTION
**Pull request by kobboi**
…ke boost dependency multilib

@EvaSDK @leio : for your approval

### CODE REVIEW

File: dev-libs/libsigc++/libsigc++-2.10.0.ebuild [hunk 1]
```diff
@@ -30,6 +30,10 @@ src_prepare() {
 			Makefile.am Makefile.in || die "sed tests failed"
 	fi
 
+	# Properly interpret --{enable,disable}-benchmark configure option (bug 599764)
+	epatch "${FILESDIR}/${P}-allow-no-benchmark.patch"
```
Comment (SoapGentoo) [inline dev-libs/libsigc++/libsigc++-2.10.0.ebuild]: use `eapply` instead, and preferably call it at the top.

File: dev-libs/libsigc++/libsigc++-2.10.0.ebuild [hunk 2]
```diff
@@ -30,6 +30,10 @@ src_prepare() {
...
+	eautoreconf
+
 	gnome2_src_prepare
 }
```
Comment (SoapGentoo) [inline dev-libs/libsigc++/libsigc++-2.10.0.ebuild]: in `multilib_src_install_all()`, replace:
```
if use doc ; then
 	insinto /usr/share/doc/${PF}
 	doins -r examples
fi
```
by
`use doc && dodoc -r examples`

File: dev-libs/libsigc++/libsigc++-2.10.0.ebuild [hunk 3]
```diff
@@ -16,10 +16,14 @@ IUSE="doc static-libs test"
...
 	doc? ( app-doc/doxygen )
-	test? ( dev-libs/boost )"
+	test? ( dev-libs/boost[${MULTILIB_USEDEP}] )"
 # Needs mm-common for eautoreconf
 
 src_prepare() {
+	# Properly interpret --{enable,disable}-benchmark configure option (bug 599764)
+	eapply "${FILESDIR}/${P}-allow-no-benchmark.patch"
+	eautoreconf
+
```
Comment (SoapGentoo) [inline dev-libs/libsigc++/libsigc++-2.10.0.ebuild]: you have to put the `eautoreconf` after the `sed`'s but before `gnome2_src_prepare`, as they affect the `Makefile.am`'s

**Comment by kobboi** [comment]
@SoapGentoo , adapted with your input. Note that in the bug report, someone does not like my patch.

**Comment by EvaSDK** [comment]
The patch is good but I would prefer direct configure sed and sending the patch upstream rather than impose eautoreconf for a simple case like this.

**Pull request closed without merging**